core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/json/UnitJsonAdapters.kt

```diff
@@ -49,6 +47,26 @@ class OffsetAdapter<T> : JsonAdapter<Offset<T>?>() {
     }
 }
 
+/**
+ * Utility class, used to put Speeds directly in json-adaptable classes. A value of type `double`
+ * will be expected, representing meters per second.
+ */
+class SpeedAdapter : JsonAdapter<Speed?>() {
+    @FromJson
+    override fun fromJson(reader: JsonReader): Speed? {
+        if (reader.peek() == JsonReader.Token.NULL) {
+            reader.skipValue()
+            return null
+        }
+        return reader.nextDouble().metersPerSecond
+    }
+
+    @ToJson
+    override fun toJson(writer: JsonWriter, value: Speed?) {
+        writer.value(if (value != null) Speed.toMetersPerSecond(value) else null)
```

Comment:
```suggestion
        writer.value(value?.metersPerSecond)
```

Not tested, but that's how it's done with other units

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/8351/commits/b4c6f6f013b673a29e906c167d7bc2d24f5f92a0